src/jit/importer.cpp

```diff
@@ -7715,7 +7715,7 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
                         unsigned   calliSlot  = lvaGrabTemp(true DEBUGARG("calli"));
                         LclVarDsc* varDsc     = &lvaTable[calliSlot];
                         varDsc->lvVerTypeInfo = tiRetVal;
-                        impAssignTempGen(calliSlot, call, clsHnd, (unsigned)CHECK_SPILL_NONE);
+                        impAssignTempGen(calliSlot, call, tiRetVal.GetClassHandle(), (unsigned)CHECK_SPILL_NONE);
```

Comment:
Yes, so my code " clsHnd  = calliSig.retTypeClass;" was bad because clsHnd had another meaning before. We can delete that as it was used only for the part of code that you changed.. 
However it doesn't explain the difference between this and sig->retTypeClass.